antarest/study/storage/variantstudy/model/command/remove_xpansion_candidate.py

```diff
@@ -43,24 +40,10 @@ class RemoveXpansionCandidate(ICommand):
     candidate_name: str
 
     @override
-    def _apply(self, study_data: FileStudy, listener: Optional[ICommandListener] = None) -> CommandOutput:
-        checks_candidate_can_be_deleted(self.candidate_name, study_data)
-        candidates = study_data.tree.get(["user", "expansion", "candidates"])
-        candidate_number = None
-        for cdt_number, cdt in candidates.items():
-            candidate = XpansionCandidate.model_validate(cdt)
-            if candidate.name == self.candidate_name:
-                candidate_number = cdt_number
-
-        if not candidate_number:
-            raise CandidateNotFoundError(f"The candidate '{self.candidate_name}' does not exist")
-
-        del candidates[candidate_number]
-        # Reorder keys of the dict
-        new_dict = {str(i): v for i, (k, v) in enumerate(candidates.items(), 1)}
-
-        study_data.tree.save(data=new_dict, url=["user", "expansion", "candidates"])
-
+    def _apply_dao(self, study_data: StudyDao, listener: Optional[ICommandListener] = None) -> CommandOutput:
+        study_data.checks_xpansion_candidate_can_be_deleted(self.candidate_name)
```

Comment:
See other command: it seems to me that we could just have a call to the delete method here ?